### DISCUSSION TITLE
Recommended ef_construct and m Values for Large Collections

### DISCUSSION THREAD

**Discussion by yanpeng7** [qdrant/qdrant#2565] [Q&A] [answered]
I currently have a collection of around 8*10^8 dim 256 image vectors and I was wondering in your experience what the optimal values for `m` and `ef_construct` are. Currently we are just using the default value of 16 and 100. If the time to compute the index is not an issue, should we just increase those 2 values and are we going to hit diminishing returns at certain values? 
Furthermore, what is the difference between setting `hnsw_config` in the top level of `create_collection` vs in the `params/vectors` struct of `create_collection`?
Thanks in advance!

**Comment by generall** [comment, marked as answer]
> . If the time to compute the index is not an issue

Usually, values beyond m=64, ef_construct=512 won't won't give much more improvement.

> Furthermore, what is the difference between setting hnsw_config in the top level of create_collection vs in the params/vectors struct of create_collection?

No difference, qdrant simply allows to specify configuration for each individual vector field, in case you are using multi-vector setup.
